File: Swift/Easy/Ransom-Note.swift
class Solution {
func canConstruct(_ ransomNote: String, _ magazine: String) -> Bool {
        var dict = [Character : Int]()
        for char in ransomNote {
            dict[char] = (dict[char] ?? 0) + 1
        }
        for char in magazine {
            if let count = dict[char] {
                if count > 1 {
                    dict[char] = count - 1
                } else {
                    dict[char] = nil
                }
            }
        }
        return dict.count == 0
    }
}

File: Python/Easy/Ransom-Note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine):
            return False 
        magazineDict = {}
        
        for char in magazine:
            if char in magazineDict:
                magazineDict[char] += 1
            else:
                magazineDict[char] = 1
        for char in ransomNote:
            if char not in magazineDict:
                return False 
            if magazineDict[char] == 0:
                return False 
            magazineDict[char] -= 1
        return True 
